php/test/Ice/proxy/Client.php

```diff
@@ -449,6 +449,8 @@ function allTests($helper)
     $c["one"] = "hello";
     $c["two"] = "world";
     $clc = $base->ice_checkedCast("::Test::MyClass", $c);
+    // TODO: the following doesn't work:
```

Comment:
Need to create an issue. This is supposed to work.

Reply:
I think the issue is how the generated checkedCast forwards the arguments:

```
public static function checkedCast($proxy, $facetOrContext=null, $context=null)
        {
            return $proxy->ice_checkedCast('::Ice::Locator', $facetOrContext, $context);
        }
```

Here the do_cast implementation in Proxy.cpp will receive 3 arguments, but the context argument is only supposed to be the second argument when there are two arguments.

Reply:
I wonder if the generated code should instead use:

```
public static function checkedCast($proxy, ...$args)
{
    return $proxy->ice_checkedCast('::Ice::Locator', ...$args);
}
```